lib/Models/CustomComponentType.js

```diff
@@ -45,16 +45,54 @@ const CustomComponentType = function(options) {
      * @type {Object[]}
      */
     this.furtherProcessing = options.furtherProcessing;
+
+    /**
+     * Gets or sets a function which checks a ReactComponent and returns a Boolean indicating if that react component
```

Comment:
Even though there's no actual React here, it's a big dodgy that this type in Models talks about ReactComponents so specifically.  Can you state it more generally in the doc?  It's ok to describe it generally and then say "in the standard TerriaJS React UI, this is a ReactComponent" or something like that.